OpenHAB Beta does not connect

### Actual behaviour
OpenHAB beta seems to be unable to connect to openHAB server (on local network). Instead, it shows only the start screen saying "Verbindung zu lokalem Server fehlgeschlagen und Sie haben keinen Server für den Fernzugriff konfiguriert". There are two buttons "Einstellungen öffenen" und "Erneut versuche", but none of them helps.
This behavoir can be reproduced with Versions 2.11.11, 2.11.10, 2.11.9. Previous versions up to 2.11.8 do not show this behaviour. Stable version 2.11.0 works on this device as expected.


### Expected behaviour
OpenHAB Beta shoud connect to openHAB server on local network and display the selected sitemap content. 

 
### Steps to reproduce
1. Update to latest Beta from Play Store
2. Open openHAB Beta


### Can you reproduce the issue in demo mode?
no.

### Environment data
#### Client
* Android version: 7.1.1
* Device model: Sony E6553
* App version: 2.11.11
* Build flavor: "Full"
* Device language: German

#### Server
* Server version: openHAB 2.5.0 Release Build
* Reverse Proxy: -
* Authentication method: None

#### openHAB Cloud
n/a

### Logs
#### App log
<details>
  <summary>Click to expand</summary>

```
-----------------------
Device information
Fingerprint: Sony/E6553/E6553:7.1.1/32.4.A.1.54/2305300192:user/release-keys
Model: E6553
Manufacturer: Sony
Brand: Sony
Device: E6553
Product: E6553
OS: 7.1.1
-----------------------

--------- beginning of main
02-03 15:41:48.893 15669 15669 W System  : ClassLoader referenced unknown path: /data/app/org.openhab.habdroid.beta-2/lib/arm64
02-03 15:41:49.006 15669 15669 I MultiDex: VM with version 2.1.0 has multidex support
02-03 15:41:49.006 15669 15669 I MultiDex: Installing application
02-03 15:41:49.006 15669 15669 I MultiDex: VM has multidex support, MultiDex support library is disabled.
02-03 15:41:49.146 15669 15669 I FirebaseInitProvider: FirebaseApp initialization successful
02-03 15:41:49.227 15669 15669 I CrashlyticsCore: Initializing Crashlytics Core 2.7.0.33
02-03 15:41:49.237 15669 15710 W Fabric  : Could not call getId on com.google.android.gms.ads.identifier.AdvertisingIdClient$Info
02-03 15:41:49.238 15669 15710 W Fabric  : Could not call isLimitAdTrackingEnabled on com.google.android.gms.ads.identifier.AdvertisingIdClient$Info
02-03 15:41:49.248 15669 15669 I CrashlyticsInitProvider: CrashlyticsInitProvider initialization successful
02-03 15:41:49.462 15669 15669 I EngineFactory: Provider GmsCore_OpenSSL not available
02-03 15:41:49.507 15669 15699 D NetworkSecurityConfig: Using Network Security Config from resource network_security_config debugBuild: false
02-03 15:41:49.525 15669 15699 E Fabric  : Settings request failed.
02-03 15:41:49.554 15669 15699 E Answers : Failed to retrieve settings
02-03 15:41:49.560 15669 15702 W CrashlyticsCore: Received null settings, skipping report submission!
02-03 15:41:49.699 15669 15669 I art     : Rejecting re-init on previously-failed class java.lang.Class<okhttp3.internal.platform.ConscryptPlatform$configureTrustManager$1>: java.lang.NoClassDefFoundError: Failed resolution of: Lorg/conscrypt/ConscryptHostnameVerifier;
02-03 15:41:49.699 15669 15669 I art     :   at okhttp3.internal.platform.android.SocketAdapter okhttp3.internal.platform.android.ConscryptSocketAdapter$Companion.buildIfSupported() (ConscryptSocketAdapter.kt:62)
02-03 15:41:49.699 15669 15669 I art     :   at void okhttp3.internal.platform.AndroidPlatform.<init>() (AndroidPlatform.kt:44)
02-03 15:41:49.699 15669 15669 I art     :   at okhttp3.internal.platform.Platform okhttp3.internal.platform.AndroidPlatform$Companion.buildIfSupported() (AndroidPlatform.kt:212)
02-03 15:41:49.699 15669 15669 I art     :   at okhttp3.internal.platform.Platform okhttp3.internal.platform.Platform$Companion.findPlatform() (Platform.kt:207)
02-03 15:41:49.699 15669 15669 I art     :   at okhttp3.internal.platform.Platform okhttp3.internal.platform.Platform$Companion.access$findPlatform(okhttp3.internal.platform.Platform$Companion) (Platform.kt:169)
02-03 15:41:49.699 15669 15669 I art     :   at void okhttp3.internal.platform.Platform.<clinit>() (Platform.kt:170)
02-03 15:41:49.699 15669 15669 I art     :   at void okhttp3.OkHttpClient.<init>(okhttp3.OkHttpClient$Builder) (OkHttpClient.kt:220)
02-03 15:41:49.699 15669 15669 I art     :   at okhttp3.OkHttpClient okhttp3.OkHttpClient$Builder.build() (OkHttpClient.kt:962)
02-03 15:41:49.699 15669 15669 I art     :   at void org.openhab.habdroid.core.connection.ConnectionFactory.<init>(android.content.Context, android.content.SharedPreferences, android.content.SharedPreferences, org.openhab.habdroid.core.connection.ConnectionManagerHelper) (ConnectionFactory.kt:110)
02-03 15:41:49.700 15669 15669 I art     :   at void org.openhab.habdroid.core.connection.ConnectionFactory$Companion.initialize(android.content.Context) (ConnectionFactory.kt:432)
02-03 15:41:49.700 15669 15669 I art     :   at void org.openhab.habdroid.core.OpenHabApplication.onCreate() (OpenHabApplication.kt:49)
02-03 15:41:49.700 15669 15669 I art     :   at void android.app.Instrumentation.callApplicationOnCreate(android.app.Application) (Instrumentation.java:1025)
02-03 15:41:49.700 15669 15669 I art     :   at void android.app.ActivityThread.handleBindApplication(android.app.ActivityThread$AppBindData) (ActivityThread.java:5548)
02-03 15:41:49.700 15669 15669 I art     :   at void android.app.ActivityThread.-wrap2(android.app.ActivityThread, android.app.ActivityThread$AppBindData) (ActivityThread.java:-1)
02-03 15:41:49.700 15669 15669 I art     :   at void android.app.ActivityThread$H.handleMessage(android.os.Message) (ActivityThread.java:1576)
02-03 15:41:49.700 15669 15669 I art     :   at void android.os.Handler.dispatchMessage(android.os.Message) (Handler.java:102)
02-03 15:41:49.700 15669 15669 I art     :   at void android.os.Looper.loop() (Looper.java:241)
02-03 15:41:49.700 15669 15669 I art     :   at void android.app.ActivityThread.main(java.lang.String[]) (ActivityThread.java:6274)
02-03 15:41:49.700 15669 15669 I art     :   at java.lang.Object java.lang.reflect.Method.invoke!(java.lang.Object, java.lang.Object[]) (Method.java:-2)
02-03 15:41:49.700 15669 15669 I art     :   at void com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run() (ZygoteInit.java:886)
02-03 15:41:49.700 15669 15669 I art     :   at void com.android.internal.os.ZygoteInit.main(java.lang.String[]) (ZygoteInit.java:776)
02-03 15:41:49.700 15669 15669 I art     : Caused by: java.lang.ClassNotFoundException: Didn't find class "org.conscrypt.ConscryptHostnameVerifier" on path: DexPathList[[zip file "/data/app/org.openhab.habdroid.beta-2/base.apk"],nativeLibraryDirectories=[/data/app/org.openhab.habdroid.beta-2/lib/arm64, /system/lib64, /vendor/lib64]]
02-03 15:41:49.700 15669 15669 I art     :   at java.lang.Class dalvik.system.BaseDexClassLoader.findClass(java.lang.String) (BaseDexClassLoader.java:56)
02-03 15:41:49.700 15669 15669 I art     :   at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String, boolean) (ClassLoader.java:380)
02-03 15:41:49.700 15669 15669 I art     :   at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String) (ClassLoader.java:312)
02-03 15:41:49.700 15669 15669 I art     :   at okhttp3.internal.platform.android.SocketAdapter okhttp3.internal.platform.android.ConscryptSocketAdapter$Companion.buildIfSupported() (ConscryptSocketAdapter.kt:62)
02-03 15:41:49.700 15669 15669 I art     :   at void okhttp3.internal.platform.AndroidPlatform.<init>() (AndroidPlatform.kt:44)
02-03 15:41:49.700 15669 15669 I art     :   at okhttp3.internal.platform.Platform okhttp3.internal.platform.AndroidPlatform$Companion.buildIfSupported() (AndroidPlatform.kt:212)
02-03 15:41:49.700 15669 15669 I art     :   at okhttp3.internal.platform.Platform okhttp3.internal.platform.Platform$Companion.findPlatform() (Platform.kt:207)
02-03 15:41:49.700 15669 15669 I art     :   at okhttp3.internal.platform.Platform okhttp3.internal.platform.Platform$Companion.access$findPlatform(okhttp3.internal.platform.Platform$Companion) (Platform.kt:169)
02-03 15:41:49.700 15669 15669 I art     :   at void okhttp3.internal.platform.Platform.<clinit>() (Platform.kt:170)
02-03 15:41:49.700 15669 15669 I art     :   at void okhttp3.OkHttpClient.<init>(okhttp3.OkHttpClient$Builder) (OkHttpClient.kt:220)
02-03 15:41:49.700 15669 15669 I art     :   at okhttp3.OkHttpClient okhttp3.OkHttpClient$Builder.build() (OkHttpClient.kt:962)
02-03 15:41:49.700 15669 15669 I art     :   at void org.openhab.habdroid.core.connection.ConnectionFactory.<init>(android.content.Context, android.content.SharedPreferences, android.content.SharedPreferences, org.openhab.habdroid.core.connection.ConnectionManagerHelper) (ConnectionFactory.kt:110)
02-03 15:41:49.700 15669 15669 I art     :   at void org.openhab.habdroid.core.connection.ConnectionFactory$Companion.initialize(android.content.Context) (ConnectionFactory.kt:432)
02-03 15:41:49.700 15669 15669 I art     :   at void org.openhab.habdroid.core.OpenHabApplication.onCreate() (OpenHabApplication.kt:49)
02-03 15:41:49.700 15669 15669 I art     :   at void android.app.Instrumentation.callApplicationOnCreate(android.app.Application) (Instrumentation.java:1025)
02-03 15:41:49.700 15669 15669 I art     :   at void android.app.ActivityThread.handleBindApplication(android.app.ActivityThread$AppBindData) (ActivityThread.java:5548)
02-03 15:41:49.700 15669 15669 I art     :   at void android.app.ActivityThread.-wrap2(android.app.ActivityThread, android.app.ActivityThread$AppBindData) (ActivityThread.java:-1)
02-03 15:41:49.700 15669 15669 I art     :   at void android.app.ActivityThread$H.handleMessage(android.os.Message) (ActivityThread.java:1576)
02-03 15:41:49.700 15669 15669 I art     :   at void android.os.Handler.dispatchMessage(android.os.Message) (Handler.java:102)
02-03 15:41:49.700 15669 15669 I art     :   at void android.os.Looper.loop() (Looper.java:241)
02-03 15:41:49.700 15669 15669 I art     :   at void android.app.ActivityThread.main(java.lang.String[]) (ActivityThread.java:6274)
02-03 15:41:49.700 15669 15669 I art     :   at java.lang.Object java.lang.reflect.Method.invoke!(java.lang.Object, java.lang.Object[]) (Method.java:-2)
02-03 15:41:49.700 15669 15669 I art     :   at void com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run() (ZygoteInit.java:886)
02-03 15:41:49.700 15669 15669 I art     :   at void com.android.internal.os.ZygoteInit.main(java.lang.String[]) (ZygoteInit.java:776)
02-03 15:41:49.700 15669 15669 I art     : 
02-03 15:41:49.858 15669 15669 D MainActivity: onCreate()
02-03 15:41:49.887 15669 15669 W art     : Before Android 4.1, method android.graphics.PorterDuffColorFilter androidx.vectordrawable.graphics.drawable.VectorDrawableCompat.updateTintFilter(android.graphics.PorterDuffColorFilter, android.content.res.ColorStateList, android.graphics.PorterDuff$Mode) would have incorrectly overridden the package-private method in android.graphics.drawable.Drawable
02-03 15:41:49.923 15669 15669 I art     : Rejecting re-init on previously-failed class java.lang.Class<androidx.core.view.ViewCompat$2>: java.lang.NoClassDefFoundError: Failed resolution of: Landroid/view/View$OnUnhandledKeyEventListener;
02-03 15:41:49.923 15669 15669 I art     :   at void androidx.core.view.ViewCompat.setOnApplyWindowInsetsListener(android.view.View, androidx.core.view.OnApplyWindowInsetsListener) (ViewCompat.java:2421)
02-03 15:41:49.923 15669 15669 I art     :   at android.view.ViewGroup androidx.appcompat.app.AppCompatDelegateImpl.createSubDecor() (AppCompatDelegateImpl.java:779)
02-03 15:41:49.923 15669 15669 I art     :   at void androidx.appcompat.app.AppCompatDelegateImpl.ensureSubDecor() (AppCompatDelegateImpl.java:659)
02-03 15:41:49.923 15669 15669 I art     :   at void androidx.appcompat.app.AppCompatDelegateImpl.setContentView(int) (AppCompatDelegateImpl.java:552)
02-03 15:41:49.923 15669 15669 I art     :   at void androidx.appcompat.app.AppCompatActivity.setContentView(int) (AppCompatActivity.java:161)
02-03 15:41:49.923 15669 15669 I art     :   at void org.openhab.habdroid.ui.MainActivity.onCreate(android.os.Bundle) (MainActivity.kt:166)
02-03 15:41:49.923 15669 15669 I art     :   at void android.app.Activity.performCreate(android.os.Bundle) (Activity.java:6720)
02-03 15:41:49.923 15669 15669 I art     :   at void android.app.Instrumentation.callActivityOnCreate(android.app.Activity, android.os.Bundle) (Instrumentation.java:1119)
02-03 15:41:49.923 15669 15669 I art     :   at android.app.Activity android.app.ActivityThread.performLaunchActivity(android.app.ActivityThread$ActivityClientRecord, android.content.Intent) (ActivityThread.java:2673)
02-03 15:41:49.923 15669 15669 I art     :   at void android.app.ActivityThread.handleLaunchActivity(android.app.ActivityThread$ActivityClientRecord, android.content.Intent, java.lang.String) (ActivityThread.java:2781)
02-03 15:41:49.923 15669 15669 I art     :   at void android.app.ActivityThread.-wrap12(android.app.ActivityThread, android.app.ActivityThread$ActivityClientRecord, android.content.Intent, java.lang.String) (ActivityThread.java:-1)
02-03 15:41:49.923 15669 15669 I art     :   at void android.app.ActivityThread$H.handleMessage(android.os.Message) (ActivityThread.java:1508)
02-03 15:41:49.923 15669 15669 I art     :   at void android.os.Handler.dispatchMessage(android.os.Message) (Handler.java:102)
02-03 15:41:49.923 15669 15669 I art     :   at void android.os.Looper.loop() (Looper.java:241)
02-03 15:41:49.923 15669 15669 I art     :   at void android.app.ActivityThread.main(java.lang.String[]) (ActivityThread.java:6274)
02-03 15:41:49.923 15669 15669 I art     :   at java.lang.Object java.lang.reflect.Method.invoke!(java.lang.Object, java.lang.Object[]) (Method.java:-2)
02-03 15:41:49.923 15669 15669 I art     :   at void com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run() (ZygoteInit.java:886)
02-03 15:41:49.923 15669 15669 I art     :   at void com.android.internal.os.ZygoteInit.main(java.lang.String[]) (ZygoteInit.java:776)
02-03 15:41:49.923 15669 15669 I art     : Caused by: java.lang.ClassNotFoundException: Didn't find class "android.view.View$OnUnhandledKeyEventListener" on path: DexPathList[[zip file "/data/app/org.openhab.habdroid.beta-2/base.apk"],nativeLibraryDirectories=[/data/app/org.openhab.habdroid.beta-2/lib/arm64, /system/lib64, /vendor/lib64]]
02-03 15:41:49.923 15669 15669 I art     :   at java.lang.Class dalvik.system.BaseDexClassLoader.findClass(java.lang.String) (BaseDexClassLoader.java:56)
02-03 15:41:49.923 15669 15669 I art     :   at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String, boolean) (ClassLoader.java:380)
02-03 15:41:49.923 15669 15669 I art     :   at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String) (ClassLoader.java:312)
02-03 15:41:49.924 15669 15669 I art     :   at void androidx.core.view.ViewCompat.setOnApplyWindowInsetsListener(android.view.View, androidx.core.view.OnApplyWindowInsetsListener) (ViewCompat.java:2421)
02-03 15:41:49.924 15669 15669 I art     :   at android.view.ViewGroup androidx.appcompat.app.AppCompatDelegateImpl.createSubDecor() (AppCompatDelegateImpl.java:779)
02-03 15:41:49.924 15669 15669 I art     :   at void androidx.appcompat.app.AppCompatDelegateImpl.ensureSubDecor() (AppCompatDelegateImpl.java:659)
02-03 15:41:49.924 15669 15669 I art     :   at void androidx.appcompat.app.AppCompatDelegateImpl.setContentView(int) (AppCompatDelegateImpl.java:552)
02-03 15:41:49.924 15669 15669 I art     :   at void androidx.appcompat.app.AppCompatActivity.setContentView(int) (AppCompatActivity.java:161)
02-03 15:41:49.924 15669 15669 I art     :   at void org.openhab.habdroid.ui.MainActivity.onCreate(android.os.Bundle) (MainActivity.kt:166)
02-03 15:41:49.924 15669 15669 I art     :   at void android.app.Activity.performCreate(android.os.Bundle) (Activity.java:6720)
02-03 15:41:49.924 15669 15669 I art     :   at void android.app.Instrumentation.callActivityOnCreate(android.app.Activity, android.os.Bundle) (Instrumentation.java:1119)
02-03 15:41:49.924 15669 15669 I art     :   at android.app.Activity android.app.ActivityThread.performLaunchActivity(android.app.ActivityThread$ActivityClientRecord, android.content.Intent) (ActivityThread.java:2673)
02-03 15:41:49.924 15669 15669 I art     :   at void android.app.ActivityThread.handleLaunchActivity(android.app.ActivityThread$ActivityClientRecord, android.content.Intent, java.lang.String) (ActivityThread.java:2781)
02-03 15:41:49.924 15669 15669 I art     :   at void android.app.ActivityThread.-wrap12(android.app.ActivityThread, android.app.ActivityThread$ActivityClientRecord, android.content.Intent, java.lang.String) (ActivityThread.java:-1)
02-03 15:41:49.924 15669 15669 I art     :   at void android.app.ActivityThread$H.handleMessage(android.os.Message) (ActivityThread.java:1508)
02-03 15:41:49.924 15669 15669 I art     :   at void android.os.Handler.dispatchMessage(android.os.Message) (Handler.java:102)
02-03 15:41:49.924 15669 15669 I art     :   at void android.os.Looper.loop() (Looper.java:241)
02-03 15:41:49.924 15669 15669 I art     :   at void android.app.ActivityThread.main(java.lang.String[]) (ActivityThread.java:6274)
02-03 15:41:49.924 15669 15669 I art     :   at java.lang.Object java.lang.reflect.Method.invoke!(java.lang.Object, java.lang.Object[]) (Method.java:-2)
02-03 15:41:49.924 15669 15669 I art     :   at void com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run() (ZygoteInit.java:886)
02-03 15:41:49.924 15669 15669 I art     :   at void com.android.internal.os.ZygoteInit.main(java.lang.String[]) (ZygoteInit.java:776)
02-03 15:41:49.924 15669 15669 I art     : 
02-03 15:41:49.949 15669 15669 I AppCompatViewInflater: app:theme is now deprecated. Please move to using android:theme instead.
02-03 15:41:50.000 15669 15669 I ViewConfigCompat: Could not find method getScaledScrollFactor() on ViewConfiguration
02-03 15:41:50.074 15669 15669 D MainActivity: Got intent: Intent { act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=org.openhab.habdroid.beta/org.openhab.habdroid.ui.MainActivity bnds=[540,936][804,1218] }
02-03 15:41:50.079 15669 15669 D MainActivity: onStart()
02-03 15:41:50.081 15669 15669 D PageConnectionHolderFragment: onStart(), started false
02-03 15:41:50.084 15669 15669 D MainActivity: onAvailableConnectionChanged()
02-03 15:41:50.088 15669 15669 D ContentController: Update to connection null (message null)
02-03 15:41:50.088 15669 15669 D PageConnectionHolderFragment: updateActiveConnections: URL list [], connection null
02-03 15:41:50.118 15669 15674 I art     : Do partial code cache collection, code=30KB, data=25KB
02-03 15:41:50.119 15669 15674 I art     : After code cache collection, code=26KB, data=25KB
02-03 15:41:50.119 15669 15674 I art     : Increasing code cache capacity to 128KB
02-03 15:41:50.120 15669 15669 D MainActivity: onPostCreate()
02-03 15:41:50.121 15669 15669 D MainActivity: onResume()
--------- beginning of events
02-03 15:41:50.135 15669 15669 I am_on_resume_called: [0,org.openhab.habdroid.ui.MainActivity,LAUNCH_ACTIVITY]
02-03 15:41:50.214 15669 15669 D Util    : normalizeUrl(): invalid URL ''
02-03 15:41:50.221 15669 15770 D ConnectionFactory: checkAvailableConnection: found connection type org.openhab.habdroid.core.connection.ConnectionManagerHelper$ConnectionType$Vpn@d710ca9
02-03 15:41:50.221 15669 15770 D AbstractConnection: Checking reachability of https://<openhab-local-address>:8443/
02-03 15:41:50.222 15669 15770 D AbstractConnection: Socket creation failed (attempt  0)
02-03 15:41:50.291 15669 15772 I Adreno  : QUALCOMM build                   : ce5aeb9, I750f7f2fa6
02-03 15:41:50.291 15669 15772 I Adreno  : Build Date                       : 03/24/17
02-03 15:41:50.291 15669 15772 I Adreno  : OpenGL ES Shader Compiler Version: XE031.09.00.03
02-03 15:41:50.291 15669 15772 I Adreno  : Local Branch                     : 
02-03 15:41:50.291 15669 15772 I Adreno  : Remote Branch                    : quic/LA.BF64.1.2.3_rb1.11
02-03 15:41:50.291 15669 15772 I Adreno  : Remote Branch                    : NONE
02-03 15:41:50.291 15669 15772 I Adreno  : Reconstruct Branch               : NOTHING
02-03 15:41:50.295 15669 15772 I OpenGLRenderer: Initialized EGL, version 1.4
02-03 15:41:50.295 15669 15772 D OpenGLRenderer: Swap behavior 1
02-03 15:41:50.330 15669 15669 D MainActivity: onCreateOptionsMenu()
02-03 15:41:50.330 15669 15669 D MainActivity: onPrepareOptionsMenu()
02-03 15:41:50.372 15669 15669 I Timeline: Timeline: Activity_idle id: android.os.BinderProxy@8143667 time:588855746
02-03 15:41:50.423 15669 15770 D AbstractConnection: Socket creation failed (attempt  1)
02-03 15:41:50.624 15669 15770 D AbstractConnection: Socket creation failed (attempt  2)
02-03 15:41:50.825 15669 15770 D AbstractConnection: Socket creation failed (attempt  3)
02-03 15:41:51.027 15669 15770 D AbstractConnection: Socket creation failed (attempt  4)
02-03 15:41:51.229 15669 15770 D AbstractConnection: Socket creation failed (attempt  5)
02-03 15:41:51.430 15669 15770 D AbstractConnection: Socket creation failed (attempt  6)
02-03 15:41:51.635 15669 15770 D AbstractConnection: Socket creation failed (attempt  7)
02-03 15:41:51.837 15669 15770 D AbstractConnection: Socket creation failed (attempt  8)
02-03 15:41:52.039 15669 15770 D AbstractConnection: Socket creation failed (attempt  9)
02-03 15:41:52.246 15669 15669 D MainActivity: onAvailableConnectionChanged()
02-03 15:41:52.246 15669 15669 D ContentController: Indicate missing configuration (resolveAttempted false)
02-03 15:41:52.247 15669 15669 D PageConnectionHolderFragment: updateActiveConnections: URL list [], connection null
02-03 15:41:52.291 15669 15669 D MainActivity: onCreateOptionsMenu()
02-03 15:41:52.292 15669 15669 D MainActivity: onPrepareOptionsMenu()
02-03 15:41:56.133 15669 15669 V BoostFramework: mAcquireFunc method = public int com.qualcomm.qti.Performance.perfLockAcquire(int,int[])
02-03 15:41:56.133 15669 15669 V BoostFramework: mReleaseFunc method = public int com.qualcomm.qti.Performance.perfLockRelease()
02-03 15:41:56.133 15669 15669 V BoostFramework: mAcquireTouchFunc method = public int com.qualcomm.qti.Performance.perfLockAcquireTouch(android.view.MotionEvent,android.util.DisplayMetrics,int,int[])
02-03 15:41:56.147 15669 15669 V BoostFramework: BoostFramework() : mPerf = com.qualcomm.qti.Performance@706ebab
02-03 15:41:56.202 15669 15669 D MainActivity: onOptionsItemSelected()
02-03 15:41:57.530 15669 15669 I Timeline: Timeline: Activity_launch_request id:org.openhab.habdroid.beta time:588862904
02-03 15:41:57.558 15669 15669 D MainActivity: onPause()
02-03 15:41:57.562 15669 15669 I am_on_paused_called: [0,org.openhab.habdroid.ui.MainActivity,handlePauseActivity]
02-03 15:41:57.618 15669 15669 I AppCompatViewInflater: app:theme is now deprecated. Please move to using android:theme instead.
02-03 15:41:59.431 15669 15669 E ExtMediaPlayer-JNI: QCMediaPlayer could not be located....
02-03 15:41:59.431 15669 15669 E MediaPlayer-JNI: JNIMediaPlayerFactory: bIsQCMediaPlayerPresent 0
02-03 15:41:59.431 15669 15669 E ExtMediaPlayer-JNI: QCMediaPlayer could not be located....
02-03 15:41:59.431 15669 15669 E MediaPlayer-JNI: JNIMediaPlayerFactory: bIsQCMediaPlayerPresent 0
02-03 15:41:59.449 15669 15669 W MediaPlayer: Couldn't open requested URI.
02-03 15:41:59.497 15669 15669 D MediaPlayer: setSubtitleAnchor in MediaPlayer
02-03 15:41:59.502 15669 15669 D Ringtone: Successfully created local player
02-03 15:41:59.515 15669 15669 D PreferencesActivity: Removing notification prefs for >= 25
02-03 15:41:59.552 15669 15669 I am_on_resume_called: [0,org.openhab.habdroid.ui.PreferencesActivity,LAUNCH_ACTIVITY]
02-03 15:41:59.562 15669 15669 I Choreographer: Skipped 118 frames!  The application may be doing too much work on its main thread.
02-03 15:41:59.688 15669 15669 D Util    : normalizeUrl(): invalid URL ''
02-03 15:41:59.703 15669 15674 I art     : Do partial code cache collection, code=53KB, data=59KB
02-03 15:41:59.704 15669 15674 I art     : After code cache collection, code=52KB, data=58KB
02-03 15:41:59.704 15669 15674 I art     : Increasing code cache capacity to 256KB
02-03 15:42:00.017 15669 15770 D ConnectionFactory: checkAvailableConnection: found connection type org.openhab.habdroid.core.connection.ConnectionManagerHelper$ConnectionType$Vpn@8cd4b4b
02-03 15:42:00.017 15669 15770 D AbstractConnection: Checking reachability of https://<openhab-local-address>:8443/
02-03 15:42:00.019 15669 15770 D AbstractConnection: Socket creation failed (attempt  0)
02-03 15:42:00.048 15669 15669 I Timeline: Timeline: Activity_idle id: android.os.BinderProxy@e892095 time:588865423
02-03 15:42:00.162 15669 15669 D MainActivity: onSaveInstanceState()
02-03 15:42:00.163 15669 15669 D ContentController: onSaveInstanceState()
02-03 15:42:00.173 15669 15669 D MainActivity: onStop()
02-03 15:42:00.174 15669 15669 D PageConnectionHolderFragment: onStop()
02-03 15:42:00.175 15669 15669 I am_on_stop_called: [0,org.openhab.habdroid.ui.MainActivity,handleStopActivity]
02-03 15:42:00.221 15669 15770 D AbstractConnection: Socket creation failed (attempt  1)
02-03 15:42:00.423 15669 15770 D AbstractConnection: Socket creation failed (attempt  2)
02-03 15:42:00.626 15669 15770 D AbstractConnection: Socket creation failed (attempt  3)
02-03 15:42:00.829 15669 15770 D AbstractConnection: Socket creation failed (attempt  4)
02-03 15:42:01.031 15669 15770 D AbstractConnection: Socket creation failed (attempt  5)
02-03 15:42:01.232 15669 15770 D AbstractConnection: Socket creation failed (attempt  6)
02-03 15:42:01.433 15669 15770 D AbstractConnection: Socket creation failed (attempt  7)
02-03 15:42:01.635 15669 15770 D AbstractConnection: Socket creation failed (attempt  8)
02-03 15:42:01.836 15669 15770 D AbstractConnection: Socket creation failed (attempt  9)
02-03 15:42:05.487 15669 15669 W ListPreference: Setting a summary with a String formatting marker is no longer supported. You should use a SummaryProvider instead.
02-03 15:42:08.064 15669 15674 I art     : Do full code cache collection, code=122KB, data=123KB
02-03 15:42:08.066 15669 15674 I art     : Starting a blocking GC JitCodeCache
02-03 15:42:08.066 15669 15674 I art     : After code cache collection, code=80KB, data=71KB
02-03 15:42:09.061 15669 15669 W ListPreference: Setting a summary with a String formatting marker is no longer supported. You should use a SummaryProvider instead.
02-03 15:42:11.239 15669 15669 I Timeline: Timeline: Activity_launch_request id:org.openhab.habdroid.beta time:588876614
02-03 15:42:11.252 15669 15669 I am_on_paused_called: [0,org.openhab.habdroid.ui.PreferencesActivity,handlePauseActivity]
02-03 15:42:11.319 15669 15669 I AppCompatViewInflater: app:theme is now deprecated. Please move to using android:theme instead.
02-03 15:42:11.325 15669 15669 I FloatingActionButton: Setting a custom background is not supported.
02-03 15:42:11.353 15669 15669 I am_on_resume_called: [0,org.openhab.habdroid.ui.LogActivity,LAUNCH_ACTIVITY]
02-03 15:42:11.424 15669 15669 D LogActivity: onCreateOptionsMenu()
02-03 15:42:11.523 15669 15772 D OpenGLRenderer: endAllActiveAnimators on 0x7f8a461c00 (RippleDrawable) with handle 0x7f8c59b2e0
02-03 15:42:11.839 15669 15669 I Timeline: Timeline: Activity_idle id: android.os.BinderProxy@c999073 time:588877213
02-03 15:42:11.871 15669 15669 I am_on_stop_called: [0,org.openhab.habdroid.ui.PreferencesActivity,handleStopActivity]
02-03 15:42:46.792 15669 15669 I am_on_paused_called: [0,org.openhab.habdroid.ui.LogActivity,handlePauseActivity]
02-03 15:43:23.120 15669 15669 I am_on_stop_called: [0,org.openhab.habdroid.ui.LogActivity,handleStopActivity]
02-03 15:44:11.878 15669 15669 I am_on_resume_called: [0,org.openhab.habdroid.ui.LogActivity,RESUME_ACTIVITY]
02-03 15:44:12.271 15669 15669 I Timeline: Timeline: Activity_idle id: android.os.BinderProxy@c999073 time:588997645
02-03 15:44:13.342 15669 15669 I am_on_paused_called: [0,org.openhab.habdroid.ui.LogActivity,handlePauseActivity]
02-03 15:44:18.111 15669 15669 I am_on_stop_called: [0,org.openhab.habdroid.ui.LogActivity,handleStopActivity]
02-03 15:44:40.449 15669 15669 I am_on_resume_called: [0,org.openhab.habdroid.ui.LogActivity,RESUME_ACTIVITY]
02-03 15:44:40.897 15669 15669 I Timeline: Timeline: Activity_idle id: android.os.BinderProxy@c999073 time:589026271
02-03 15:44:43.110 15669 15669 I am_on_paused_called: [0,org.openhab.habdroid.ui.LogActivity,handlePauseActivity]
02-03 15:44:49.525 15669 15669 I am_on_resume_called: [0,org.openhab.habdroid.ui.LogActivity,RESUME_ACTIVITY]
02-03 15:44:49.531 15669 15674 I art     : Do partial code cache collection, code=122KB, data=123KB
02-03 15:44:49.532 15669 15674 I art     : After code cache collection, code=119KB, data=121KB
02-03 15:44:49.532 15669 15674 I art     : Increasing code cache capacity to 512KB
02-03 15:44:49.952 15669 15669 I Timeline: Timeline: Activity_idle id: android.os.BinderProxy@c999073 time:589035327
02-03 15:44:50.839 15669 15669 I am_on_paused_called: [0,org.openhab.habdroid.ui.LogActivity,handlePauseActivity]
02-03 15:44:50.931 15669 15669 I am_on_stop_called: [0,org.openhab.habdroid.ui.LogActivity,handleStopActivity]
02-03 15:45:08.139 15669 15669 I am_on_resume_called: [0,org.openhab.habdroid.ui.LogActivity,RESUME_ACTIVITY]
```

</details>


#### openHAB Server log
<details>
  <summary>Click to expand</summary>

```
Not required.
If you have installed openHAB via `apt-get`, the log is located under `/var/log/openhab2/openhab.log`
```

</details>

--- a/mobile/src/main/java/org/openhab/habdroid/core/connection/ConnectionFactory.kt
+++ b/mobile/src/main/java/org/openhab/habdroid/core/connection/ConnectionFactory.kt
@@ -16,7 +16,6 @@ package org.openhab.habdroid.core.connection
 import android.app.Activity
 import android.content.Context
 import android.content.SharedPreferences
-import android.net.Network
 import android.security.KeyChain
 import android.security.KeyChainException
 import android.util.Log
@@ -306,54 +305,51 @@ class ConnectionFactory internal constructor(
     }
 
     private fun checkAvailableConnection(local: Connection?, remote: Connection?): Connection {
-        fun handleLocalNetworkType(network: Network?): Connection {
-            if (local is DefaultConnection) {
-                local.network = network
-            }
-            return when {
-                // If local URL is configured and reachable
-                local?.checkReachabilityInBackground() == true -> {
-                    Log.d(TAG, "Connecting to local URL")
-                    local
+        val available = connectionHelper.currentConnections
+            .sortedBy { type -> when (type) {
+                is ConnectionManagerHelper.ConnectionType.Vpn -> 1
+                is ConnectionManagerHelper.ConnectionType.Ethernet -> 2
+                is ConnectionManagerHelper.ConnectionType.Wifi -> 3
+                is ConnectionManagerHelper.ConnectionType.Mobile -> 4
+                is ConnectionManagerHelper.ConnectionType.Unknown -> 5
+            } }
+
+        Log.d(TAG, "checkAvailableConnection: found types $available")
+        if (available.isEmpty()) {
+            Log.e(TAG, "Network is not available")
+            throw NetworkNotAvailableException()
+        }
+
+        if (local != null) {
+            val localCandidates = available
+                .filter { type ->
+                    type is ConnectionManagerHelper.ConnectionType.Wifi
+                        || type is ConnectionManagerHelper.ConnectionType.Ethernet
+                        || type is ConnectionManagerHelper.ConnectionType.Vpn
                 }
-                remote != null -> {
-                    // If local URL is not reachable or not configured, use remote URL
-                    Log.d(TAG, "Connecting to remote URL")
-                    remote
+            for (type in localCandidates) {
+                if (local is DefaultConnection) {
+                    local.network = type.network
                 }
-                else -> throw NoUrlInformationException(true)
+                if (local.checkReachabilityInBackground()) {
+                    Log.d(TAG, "Connecting to local URL via $type")
+                    return local
+                }
             }
         }
 
-        val type = connectionHelper.currentConnection
-        Log.d(TAG, "checkAvailableConnection: found connection type $type")
-
-        return when (type) {
-            is ConnectionManagerHelper.ConnectionType.None -> {
-                Log.e(TAG, "Network is not available")
-                throw NetworkNotAvailableException()
-            }
-            // If we are on a mobile network go directly to remote URL from settings
-            is ConnectionManagerHelper.ConnectionType.Mobile -> {
-                remote ?: throw NoUrlInformationException(false)
-            }
-            // Else if we are on Wifi, Ethernet or VPN network, check whether local connection
-            // is available and use remote connection if it isn't available
-            is ConnectionManagerHelper.ConnectionType.Wifi -> {
-                handleLocalNetworkType(type.network)
-            }
-            is ConnectionManagerHelper.ConnectionType.Ethernet -> {
-                handleLocalNetworkType(type.network)
-            }
-            is ConnectionManagerHelper.ConnectionType.Vpn -> {
-                handleLocalNetworkType(type.network)
-            }
-            // Else we treat other networks types as unsupported
-            else -> {
-                Log.e(TAG, "Network type $type is unsupported")
-                throw NetworkNotSupportedException()
-            }
+        if (available[0] is ConnectionManagerHelper.ConnectionType.Unknown) {
+            Log.e(TAG, "Network type ${available[0]} is unsupported")
+            throw NetworkNotSupportedException()
         }
+
+        if (remote != null) {
+            // If local URL is not reachable or not configured, use remote URL
+            Log.d(TAG, "Connecting to remote URL")
+            return remote
+        }
+
+        throw NoUrlInformationException(true)
     }
 
     private class ClientKeyManager(context: Context, private val alias: String?) : X509KeyManager {

--- a/mobile/src/main/java/org/openhab/habdroid/core/connection/ConnectionFactory.kt
+++ b/mobile/src/main/java/org/openhab/habdroid/core/connection/ConnectionFactory.kt
@@ -323,9 +323,9 @@ class ConnectionFactory internal constructor(
         if (local != null) {
             val localCandidates = available
                 .filter { type ->
-                    type is ConnectionManagerHelper.ConnectionType.Wifi
-                        || type is ConnectionManagerHelper.ConnectionType.Ethernet
-                        || type is ConnectionManagerHelper.ConnectionType.Vpn
+                    type is ConnectionManagerHelper.ConnectionType.Wifi ||
+                        type is ConnectionManagerHelper.ConnectionType.Ethernet ||
+                        type is ConnectionManagerHelper.ConnectionType.Vpn
                 }
             for (type in localCandidates) {
                 if (local is DefaultConnection) {

--- a/mobile/src/main/java/org/openhab/habdroid/core/connection/ConnectionManagerHelper.kt
+++ b/mobile/src/main/java/org/openhab/habdroid/core/connection/ConnectionManagerHelper.kt
@@ -30,18 +30,17 @@ import kotlinx.coroutines.launch
 
 typealias ConnectionChangedCallback = () -> Unit
 interface ConnectionManagerHelper {
-    val currentConnection: ConnectionType
+    val currentConnections: List<ConnectionType>
     var changeCallback: ConnectionChangedCallback?
 
     fun shutdown()
 
-    sealed class ConnectionType {
-        class None : ConnectionType()
-        class Wifi(val network: Network?) : ConnectionType()
-        class Ethernet(val network: Network?) : ConnectionType()
-        class Mobile(val network: Network?) : ConnectionType()
-        class Unknown(val network: Network?) : ConnectionType()
-        class Vpn(val network: Network?) : ConnectionType()
+    sealed class ConnectionType constructor(val network: Network?) {
+        class Wifi(network: Network?) : ConnectionType(network)
+        class Ethernet(network: Network?) : ConnectionType(network)
+        class Mobile(network: Network?) : ConnectionType(network)
+        class Unknown(network: Network?) : ConnectionType(network)
+        class Vpn(network: Network?) : ConnectionType(network)
     }
 
     companion object {
@@ -57,19 +56,19 @@ interface ConnectionManagerHelper {
     private class HelperApi22 constructor(context: Context) :
         ConnectionManagerHelper, ChangeCallbackHelperApi25(context) {
         private val typeHelper = NetworkTypeHelperApi22(context)
-        override val currentConnection: ConnectionType get() = typeHelper.currentConnection
+        override val currentConnections: List<ConnectionType> get() = typeHelper.currentConnections
     }
 
     private class HelperApi23To25 constructor(context: Context) :
         ConnectionManagerHelper, ChangeCallbackHelperApi25(context) {
         private val typeHelper = NetworkTypeHelperApi23(context)
-        override val currentConnection: ConnectionType get() = typeHelper.currentConnection
+        override val currentConnections: List<ConnectionType> get() = typeHelper.currentConnections
     }
 
     private class HelperApi26 constructor(context: Context) :
         ConnectionManagerHelper, ChangeCallbackHelperApi26(context) {
         private val typeHelper = NetworkTypeHelperApi26(context)
-        override val currentConnection: ConnectionType get() = typeHelper.currentConnection
+        override val currentConnections: List<ConnectionType> get() = typeHelper.currentConnections
     }
 
     private open class ChangeCallbackHelperApi25 constructor(context: Context) : BroadcastReceiver() {
@@ -154,18 +153,16 @@ interface ConnectionManagerHelper {
     @Suppress("DEPRECATION")
     private class NetworkTypeHelperApi22 constructor(context: Context) {
         private val connectivityManager = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
-        val currentConnection: ConnectionType get() {
-            val activeConnectionTypes = connectivityManager.allNetworkInfo
+        val currentConnections: List<ConnectionType> get() {
+            return connectivityManager.allNetworkInfo
                     .filter { info -> info.isConnected }
-                    .map { info -> info.type }
-            return when {
-                activeConnectionTypes.isEmpty() -> ConnectionType.None()
-                ConnectivityManager.TYPE_VPN in activeConnectionTypes -> ConnectionType.Vpn(null)
-                ConnectivityManager.TYPE_WIFI in activeConnectionTypes -> ConnectionType.Wifi(null)
-                ConnectivityManager.TYPE_ETHERNET in activeConnectionTypes -> ConnectionType.Ethernet(null)
-                ConnectivityManager.TYPE_MOBILE in activeConnectionTypes -> ConnectionType.Mobile(null)
-                else -> ConnectionType.Unknown(null)
-            }
+                    .map { info -> when (info.type) {
+                        ConnectivityManager.TYPE_VPN -> ConnectionType.Vpn(null)
+                        ConnectivityManager.TYPE_WIFI -> ConnectionType.Wifi(null)
+                        ConnectivityManager.TYPE_ETHERNET -> ConnectionType.Ethernet(null)
+                        ConnectivityManager.TYPE_MOBILE -> ConnectionType.Mobile(null)
+                        else -> ConnectionType.Unknown(null)
+                    } }
         }
     }
 
@@ -173,40 +170,25 @@ interface ConnectionManagerHelper {
     @Suppress("DEPRECATION")
     private class NetworkTypeHelperApi23 constructor(context: Context) {
         private val connectivityManager = context.getSystemService(ConnectivityManager::class.java)!!
-        val currentConnection: ConnectionType get() {
-            val knownConnectionTypesByPrio = listOf(
-                ConnectivityManager.TYPE_VPN,
-                ConnectivityManager.TYPE_WIFI,
-                ConnectivityManager.TYPE_ETHERNET,
-                ConnectivityManager.TYPE_MOBILE
-            )
-            val active = connectivityManager.allNetworks
+        val currentConnections: List<ConnectionType> get() {
+            return connectivityManager.allNetworks
                     .map { network -> network to connectivityManager.getNetworkInfo(network) }
                     .filter { (_, info) -> info?.isConnected == true }
                     // info can not be null here, as the condition in the line above covers that case already
-                    .map { (network, info) -> info!!.type to network }
-                    // sort the list by connection type priority, putting irrelevant networks at the end
-                    .sortedBy { (type, _) ->
-                        val index = knownConnectionTypesByPrio.indexOf(type)
-                        if (index < 0) Integer.MAX_VALUE else index
-                    }
-                    .firstOrNull()
-
-            return when (active?.first) {
-                null -> ConnectionType.None()
-                ConnectivityManager.TYPE_VPN -> ConnectionType.Vpn(active.second)
-                ConnectivityManager.TYPE_WIFI -> ConnectionType.Wifi(active.second)
-                ConnectivityManager.TYPE_ETHERNET -> ConnectionType.Ethernet(active.second)
-                ConnectivityManager.TYPE_MOBILE -> ConnectionType.Mobile(active.second)
-                else -> ConnectionType.Unknown(active.second)
-            }
+                    .map { (network, info) -> when (info!!.type) {
+                        ConnectivityManager.TYPE_VPN -> ConnectionType.Vpn(network)
+                        ConnectivityManager.TYPE_WIFI -> ConnectionType.Wifi(network)
+                        ConnectivityManager.TYPE_ETHERNET -> ConnectionType.Ethernet(network)
+                        ConnectivityManager.TYPE_MOBILE -> ConnectionType.Mobile(network)
+                        else -> ConnectionType.Unknown(network)
+                    } }
         }
     }
 
     @TargetApi(26)
     private class NetworkTypeHelperApi26 constructor(context: Context) {
         private val connectivityManager = context.getSystemService(ConnectivityManager::class.java)!!
-        val currentConnection: ConnectionType get() {
+        val currentConnections: List<ConnectionType> get() {
             val knownTransportsByPrio = listOf(
                 NetworkCapabilities.TRANSPORT_VPN,
                 NetworkCapabilities.TRANSPORT_WIFI,
@@ -217,25 +199,20 @@ interface ConnectionManagerHelper {
             fun findMostRelevantTransport(caps: NetworkCapabilities?): Int? =
                 knownTransportsByPrio.firstOrNull { transport -> caps?.hasTransport(transport) == true }
 
-            val active = connectivityManager.allNetworks
+            return connectivityManager.allNetworks
                 .map { network -> network to connectivityManager.getNetworkCapabilities(network) }
                 .filter { (_, caps) -> caps?.isUsable() == true }
                 .map { (network, caps) -> network to findMostRelevantTransport(caps) }
                 // filter out irrelevant networks
                 .filter { (_, transport) -> transport != null }
-                // sort remaining relevant networks by their priority
-                .sortedBy { (_, transport) -> knownTransportsByPrio.indexOf(transport) }
-                .firstOrNull()
-
-            return when (active?.second) {
-                null -> ConnectionType.None()
-                NetworkCapabilities.TRANSPORT_VPN -> ConnectionType.Vpn(active.first)
-                NetworkCapabilities.TRANSPORT_WIFI,
-                NetworkCapabilities.TRANSPORT_WIFI_AWARE -> ConnectionType.Wifi(active.first)
-                NetworkCapabilities.TRANSPORT_ETHERNET -> ConnectionType.Ethernet(active.first)
-                NetworkCapabilities.TRANSPORT_CELLULAR -> ConnectionType.Mobile(active.first)
-                else -> ConnectionType.Unknown(active.first)
-            }
+                .map { (network, transport) -> when (transport) {
+                    NetworkCapabilities.TRANSPORT_VPN -> ConnectionType.Vpn(network)
+                    NetworkCapabilities.TRANSPORT_WIFI,
+                    NetworkCapabilities.TRANSPORT_WIFI_AWARE -> ConnectionType.Wifi(network)
+                    NetworkCapabilities.TRANSPORT_ETHERNET -> ConnectionType.Ethernet(network)
+                    NetworkCapabilities.TRANSPORT_CELLULAR -> ConnectionType.Mobile(network)
+                    else -> ConnectionType.Unknown(network)
+                } }
         }
     }
 }

--- a/mobile/src/main/java/org/openhab/habdroid/core/connection/ConnectionManagerHelper.kt
+++ b/mobile/src/main/java/org/openhab/habdroid/core/connection/ConnectionManagerHelper.kt
@@ -36,11 +36,11 @@ interface ConnectionManagerHelper {
     fun shutdown()
 
     sealed class ConnectionType constructor(val network: Network?) {
-        class Wifi(network: Network?) : ConnectionType(network)
         class Ethernet(network: Network?) : ConnectionType(network)
         class Mobile(network: Network?) : ConnectionType(network)
         class Unknown(network: Network?) : ConnectionType(network)
         class Vpn(network: Network?) : ConnectionType(network)
+        class Wifi(network: Network?) : ConnectionType(network)
     }
 
     companion object {

--- a/mobile/src/test/java/org/openhab/habdroid/core/connection/ConnectionFactoryTest.kt
+++ b/mobile/src/test/java/org/openhab/habdroid/core/connection/ConnectionFactoryTest.kt
@@ -165,7 +165,7 @@ class ConnectionFactoryTest {
     @Test(expected = NetworkNotAvailableException::class)
     @Throws(ConnectionException::class)
     fun testGetAnyConnectionNoNetwork() {
-        mockConnectionHelper.update(ConnectionManagerHelper.ConnectionType.None())
+        mockConnectionHelper.update(null)
         updateAndWaitForConnections()
         ConnectionFactory.usableConnection
     }
@@ -230,12 +230,11 @@ class ConnectionFactoryTest {
 
     private inner class MockConnectionHelper : ConnectionManagerHelper {
         override var changeCallback: ConnectionChangedCallback? = null
-        private var currentType: ConnectionManagerHelper.ConnectionType =
-            ConnectionManagerHelper.ConnectionType.Unknown(null)
-        override val currentConnection: ConnectionManagerHelper.ConnectionType get() = currentType
+        private var currentTypes: List<ConnectionManagerHelper.ConnectionType> = emptyList()
+        override val currentConnections: List<ConnectionManagerHelper.ConnectionType> get() = currentTypes
 
-        fun update(type: ConnectionManagerHelper.ConnectionType) {
-            currentType = type
+        fun update(type: ConnectionManagerHelper.ConnectionType?) {
+            currentTypes = if (type != null) listOf(type) else emptyList()
             changeCallback?.invoke()
         }
         override fun shutdown() {}

